clang/lib/CConv/DeclRewriter.cpp

```diff
@@ -175,6 +179,130 @@ void DeclRewriter::rewriteParmVarDecl(ParmVarDeclReplacement *N) {
     }
 }
 
+void DeclRewriter::rewriteFieldDecl(FieldDeclReplacement *N, RSet &ToRewrite) {
```

Comment:
Copy & pasting the same code for `VarDecl` seems bad. Small changes to one will have to be duplicated somewhere else. @jackastner Any ideas how this stuff can get combined together, to avoid code duplication?

Reply:
I'll try to make it generic, was trying to avoid making the code unclear since their are some differences it the handling needing for variables. (Ie: variables can have initializers whereas fields cannot)

Reply:
OK. But from a quick glance it seems that there is much more code in common than not.

Is there a way we can use the class hierarchy for this? Make FieldDecls a subclass of VarDecls, for purposes of rewriting?

Reply:
They are, and I'm working through it now. Shouldn't take long.